htseq2multx.pl

```diff
@@ -0,0 +1,976 @@
+                        at the beginning)
+
+Output Options:
+  --prefix PREFIX       Prefix for output files
+  --suffix SUFFIX       Suffix for output files (default based on --format)
+  --galaxy              Produce "Galaxy safe" filenames by removing
+                        underscores (default: False)
+  --nosanitize          Do not produce "safe" filenames by replacing unusual
+                        characters in the supplied prefix and sample IDs with
+                        underscores. (default: False)
+  -v, --verbose         verbose output
+  --gzipout             Output files in compressed gzip format (default is
+                        uncompressed)
+  --split_all           Split all input files, including index read files (by
+                        default, index read files are not split unless all
+                        read files are index files)
+
+Input format:
+  --format FORMAT       Specify format for sequence files (fasta or fastq)
+  --gzipin              Assume input files are in gzip format, despite file
+                        extension (default is auto based on input file
+                        extension)
+
+Sequence Inputs:
+  FILE                  A series of 1 or more [optionally zipped] fastq files.
+
+* Supply --help twice to see differences between barcode_splitter and this tool.
+END_LONG
+
+    print($short_msg);
+    unless($short_only)
+      {print($long_msg)}
+  }
+
+sub help2
+  {
+    my $msg = << 'END_HELP2';
+
+The following extra and modified options apply specifically to the htseq2multx.pl wrapper (as opposed to barcode_splitter).
+
+arguments with behavioral differences:
+  --bcfile FILE         barcode_splitter requires all barcodes in a single
+                        column to be the same length, however fastq-multx
+                        supports variable length barcodes, thus the length
+                        consistency check is not performed and no errors about
+                        different barcode lengths are issued.
+
+optional arguments:
+  -h, --help            Supply twice to show this message and exit.
+  --version             Supply once to get the barcode_splitter version that
+                        this wrapper emulates.  Supply twice to get a more
+                        verbose version of the barcode_splitter version that
+                        this wrapper emulates and the version of this wrapper.
+
+configuration arguments:
+  --fastq-multx PATH    Path to the fastq-multx installation.  Default is to use
+                        the PATH in the user's environment.
+  --debug               Supply this flag to output the fastq-multx command on
+                        standard error and not delete the temporary barcode
+                        file.
+END_HELP2
+
+    print($msg);
+  }
+
+sub getFastqMultxCommand
+  {
+    my $command =
+      "$fastq_multx -x -d 1 -B '$bcfilefqmx' -m $mismatches -M $mismatches";
```

Comment:
```suggestion
    # -x          Don't trim barcodes off before writing out destination
    # -d N        Require a minimum distance of N between the best and next best (2)
    # -B BCFIL    Use barcodes from BCFIL, no determination step, codes in <read1.fq>
    # -m N        Allow N mismatches in union of all indexes, unless -M is supplied. (1)
    # -M M        Allow N,M mismatches in indexes 1,2 respectively (see -m N) (not used)
    my $command =
      "$fastq_multx -x -d 1 -B '$bcfilefqmx' -m $mismatches -M $mismatches";
```

Reply:
I like this idea.  I think though that to be accurate to the user's version, maybe I should create a way to output the  executable's usage...? Although perhaps I should output the version that this command was based on...  Hmmm... I'll think about it.

Reply:
There really won't be users of this other than us, I just think it's helpful to document what these options are.

Reply:
Right. Well, we could just run fastq-multx to see the usage. But I'll add a comment.

- [ ] Comment on the fastq-multx options used